algorithms task (1)/student/include/impl.h

```diff
@@ -11,4 +11,12 @@
   * @return consecutive range
 */
 template<class FI, class Comparator = std::equal_to<typename std::iterator_traits<FI>::value_type>>
-std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {});
+std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {})
+{
+    auto firstEqual = std::adjacent_find(start, end, comp);
+    auto lastEqual = std::find_if_not(firstEqual, end, [&firstEqual](typename std::iterator_traits<FI>::value_type element)
+    {
+        return element == *firstEqual;
```

Comment:
**for comparison, Comparator function is used instead of non-generic equality operator**